### PR TITLE
Add encryption and authentication

### PR DESCRIPTION
**Pull request by scholzj**
This PR adds a new docu section about Kafka encryption and authentication.

### CODE REVIEW

File: books/proc-kafka-deleting-scram-users.adoc [hunk 1]
```diff
@@ -0,0 +1,30 @@
+// Module included in the following assemblies:
+//
+// assembly-kafka-encryption-and-authentication.adoc
+
+[id='proc-kafka-deleting-scram-users-{context}']
+
+= Deleting SASL SCRAM users
+
+This procedure will show you remove users when using SASL SCRAM authentication.
```
Comment (ppatierno) [inline books/proc-kafka-deleting-scram-users.adoc]: "show you how to ..."

File: books/proc-kafka-enable-tls-client-authentication.adoc [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+This procedure will show you how to enable TLS client authentication in Kafka brokers.
+
+.Prerequisites
+
+* AMQ Streams is xref:proc-installing-amq-streams-{context}[installed] on all hosts which will be used as Kafka brokers.
+* TLS encryption is xref:proc-kafka-enable-tls-encryption-{context}[enabled].
+
+.Enabling TLS client authentication
+
+. Prepare a JKS truststore containing the public key of the CA used to sing the user certificates.
```
Comment (ppatierno) [inline books/proc-kafka-enable-tls-client-authentication.adoc]: until now we have used certification authority, now we are using CA. Should we clarify that? I mean using certification authority here or at the beginning of this section making the point that CA is abbreviation for certification authority.

File: books/con-kafka-tls-encryption.adoc [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+ssl.keystore.location=/path/to/keystore/server-1.jks
+ssl.keystore.password=123456
+
+# Different configuration for listener REPLICATION
+listener.name.replication.ssl.keystore.location=/path/to/keystore/server-1.jks
+listener.name.replication.ssl.keystore.password=123456
+----
+
+In addition to the main TLS configuration options described above, Kafka supports many options for fine-tuning the TLS configuration.
+For example, to enable of disable TLS / SSL protocols or cipher suites:
```
Comment (tombentley) [inline books/con-kafka-tls-encryption.adoc]: typo: of → or

File: books/con-kafka-tls-encryption.adoc [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+
+`ssl.cipher.suites`::
+List of enabled cipher suites.
+Each cipher suite is a combination of authentication, encryption, MAC and key exchange algorithms used for the TLC connection.
```
Comment (tombentley) [inline books/con-kafka-tls-encryption.adoc]: typo: TLC → TLS

File: books/proc-kafka-enable-plain-authentication.adoc [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+
+This procedure will show you how to enable SASL PLAIN authentication in Kafka brokers.
+
+.Prerequisites
+
+* AMQ Streams is xref:proc-installing-amq-streams-{context}[installed] on all hosts which will be used as Kafka brokers.
+
+.Enabling SASL PLAIN authentication
+
+. Edit of create the `/opt/kafka/config/jaas.conf` JAAS configuration file.
```
Comment (tombentley) [inline books/proc-kafka-enable-plain-authentication.adoc]: typo: of → or

File: books/proc-kafka-enable-scram-authentication.adoc [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+
+This procedure will show you how to enable SASL SCRAM authentication in Kafka brokers.
+
+.Prerequisites
+
+* AMQ Streams is xref:proc-installing-amq-streams-{context}[installed] on all hosts which will be used as Kafka brokers.
+
+.Enabling SASL SCRAM authentication
+
+. Edit of create the `/opt/kafka/config/jaas.conf` JAAS configuration file.
```
Comment (tombentley) [inline books/proc-kafka-enable-scram-authentication.adoc]: typo: of → or

File: books/proc-kafka-enable-scram-authentication.adoc [hunk 7]
```diff
@@ -0,0 +1,59 @@
...
+----
+KafkaServer {
+    org.apache.kafka.common.security.scram.ScramLoginModule required;
+};
+----
+
+. Edit the `/opt/kafka/config/server.properties` Kafka configuration file on all cluster nodes for the following:
++
+* Change the `listener.security.protocol.map` field to specify the `SASL_PLAINTEXT` or `SASL_SSL` protocol for the listener where you want to use SASL SCRAM authentication.
+* Set the `sasl.enabled.mechanisms` option to `SCRAM-SHA-512`.
```
Comment (tombentley) [inline books/proc-kafka-enable-scram-authentication.adoc]: or -256

File: books/proc-kafka-enable-tls-client-authentication.adoc (hunk 2 above)
Comment (tombentley) [reply books/proc-kafka-enable-tls-client-authentication.adoc]: typo: sing → sign

File: books/proc-kafka-enable-tls-encryption.adoc [hunk 8]
```diff
@@ -0,0 +1,42 @@
...
+This procedure will show you how to enable encryption in Kafka brokers.
+
+.Prerequisites
+
+* AMQ Streams is xref:proc-installing-amq-streams-{context}[installed] on all hosts which will be used as Kafka brokers.
+
+.Enabling TLS encryption
+
+. Generate TLS certificates for all Kafka brokers in your cluster.
+The certificates should have their advertised and bootstrap addresses in their Common Name or Subject Alternative Name.
```
Comment (tombentley) [inline books/proc-kafka-enable-tls-encryption.adoc]: Should we include examples using `openssl`?

File: books/assembly-kafka-encryption-and-authentication.adoc [hunk 9]
```diff
@@ -0,0 +1,25 @@
+= Encryption and authentication
+
+Kafka supports TLS for encrypting the communication with Kafka clients.
+Additionally, it supports two types of authentication:
+
+* TLS client authentication based on TLC client certificates
+* SASL Authentication based on username and password
```
[the remark this answers is not in the document]
Comment (scholzj) [reply books/assembly-kafka-encryption-and-authentication.adoc]: It certainly looks like we don't have time to do that for beta.

File: books/con-kafka-authentication.adoc [hunk 10]
```diff
@@ -0,0 +1,199 @@
...
+The SASL mechanism used for authentication using Kerberos is called `GSSAPI`. To configure Kerberos SASL authentication,
+the following configuration should be added to the JAAS configuration file:
+
+[source]
+KafkaServer {
+    com.sun.security.auth.module.Krb5LoginModule required
+    useKeyTab=true
+    storeKey=true
+    keyTab="/etc/security/keytabs/kafka_server.keytab"
+    principal="kafka/kafka1.hostname.com@EXAMPLE.COM";
```
[the remark this answers is not in the document]
Comment (tombentley) [reply books/con-kafka-authentication.adoc]: Yes: https://web.mit.edu/kerberos/krb5-1.5/krb5-1.5.4/doc/krb5-user/What-is-a-Kerberos-Principal_003f.html

File: books/proc-kafka-enable-tls-encryption.adoc (hunk 8 above)
Comment (scholzj) [reply books/proc-kafka-enable-tls-encryption.adoc]: No, there are many ways how to do it, I do not think we need to list any examples for that.

File: books/con-kafka-authentication.adoc (hunk 10 above)
[the remark this answers is not in the document]
Comment (scholzj) [reply books/con-kafka-authentication.adoc]: Ok, I added it there.

File: books/con-kafka-authentication.adoc [hunk 11]
```diff
@@ -0,0 +1,199 @@
...
+TLS client authentication is switched off. (Default value)
+
+`requested`::
+TLS client authentication is optional.
+Clients will be asked to authenticate using TLS client certificate but they can choose not to.
+
+`required`::
+Clients are required to authenticate using TLS client certificate.
+
+When a client authenticates using TLS client authentication, the authenticated principal name is the distinguished name from the authenticated client certificate.
```
[the remark this answers is not in the document]
Comment (scholzj) [reply books/con-kafka-authentication.adoc]: I added a simple example. Thanks.

File: books/con-kafka-authentication.adoc [hunk 12]
```diff
@@ -0,0 +1,199 @@
...
+    org.apache.kafka.common.security.plain.PlainLoginModule required
+    username="admin"
+    password="123456"
+    user_admin="123456"
+    user_user1="123456"
+    user_user2="123456";
+};
+----
+
+SCRAM authentication in Kafka consists of two mechanisms: `SCRAM-SHA-256` and `SCRAM-SHA-512`.
```
[the remark this answers is not in the document]
Comment (tombentley) [reply books/con-kafka-authentication.adoc]: "Mechanism" has a meaning within the SASL spec, for... well SASL mechanisms. Earlier you talked about TLS and SASL as the two supported "authentication mechanisms" (using the word in a non-SASL sense). Here you're talking about two other mechanisms. I personally don't think it's confusing, but I can imagine there would be readers who were confused by using mechanism in two different ways like this.

I don't think this is a big deal, but you could reword [this line](https://github.com/rh-messaging/amq-streams-docs/pull/13/files#diff-f97f13ccd89863008550cd82e35d2747R9) to read "Kafka supports two methods of authentication." or similar and then use the word "mechanism" is a SASL-specific sense.

[the remark this answers is not in the document]
Comment (scholzj) [reply books/con-kafka-authentication.adoc]: Ahh, ok. Understand. I think it makes sense. I fixed it.

File: books/proc-kafka-adding-scram-users.adoc [hunk 13]
```diff
@@ -0,0 +1,30 @@
+// Module included in the following assemblies:
+//
+// assembly-kafka-encryption-and-authentication.adoc
+
+[id='proc-kafka-adding-scram-users-{context}']
+
+= Adding SASL SCRAM users
+
+This procedure will show you how to add new users for authentication using SASL SCRAM.
```
Comment (dmuntima) [inline books/proc-kafka-adding-scram-users.adoc]: will show you how to >  describes how to

File: books/proc-kafka-adding-scram-users.adoc [hunk 14]
```diff
@@ -0,0 +1,30 @@
...
+
+.Prerequisites
+
+* AMQ Streams is xref:proc-installing-amq-streams-{context}[installed] on all hosts which will be used as Kafka brokers.
+* SASL SCRAM authentication is xref:proc-kafka-enable-scram-authentication-{context}[enabled].
+
+.Adding SASL SCRAM users
```
Comment (dmuntima) [inline books/proc-kafka-adding-scram-users.adoc]: .Adding SASL SCRAM users > .Procedure

File: books/proc-kafka-adding-scram-users.adoc [hunk 15]
```diff
@@ -0,0 +1,30 @@
...
+
+. Use the utility `kafka-configs.sh` to add new SASL SCRAM users.
```
Comment (dmuntima) [inline books/proc-kafka-adding-scram-users.adoc]: - Use the utility `kafka-configs.sh`  > Use the `kafka-configs.sh` tool
- - Change numbered step to a bulleted list as it is single-step procedure

File: books/proc-kafka-deleting-scram-users.adoc [hunk 16]
```diff
@@ -0,0 +1,30 @@
+// Module included in the following assemblies:
+//
+// assembly-kafka-encryption-and-authentication.adoc
+
+[id='proc-kafka-deleting-scram-users-{context}']
+
+= Deleting SASL SCRAM users
+
+This procedure will show you how to remove users when using SASL SCRAM authentication.
```
Comment (dmuntima) [inline books/proc-kafka-deleting-scram-users.adoc]: will show you how to > describes how to

File: books/proc-kafka-deleting-scram-users.adoc [hunk 17]
```diff
@@ -0,0 +1,30 @@
...
+
+.Prerequisites
+
+* AMQ Streams is xref:proc-installing-amq-streams-{context}[installed] on all hosts which will be used as Kafka brokers.
+* SASL SCRAM authentication is xref:proc-kafka-enable-scram-authentication-{context}[enabled].
+
+.Adding SASL SCRAM users
```
Comment (dmuntima) [inline books/proc-kafka-deleting-scram-users.adoc]: .Adding SASL SCRAM users > .Procedure

File: books/proc-kafka-deleting-scram-users.adoc [hunk 18]
```diff
@@ -0,0 +1,30 @@
...
+
+. Use the utility `kafka-configs.sh` to delete SASL SCRAM users.
```
Comment (dmuntima) [inline books/proc-kafka-deleting-scram-users.adoc]: - Use the utility `kafka-configs.sh` > Use the `kafka-configs.sh` tool
- Change numbered step to a bulleted list as it is single-step procedure

File: books/proc-kafka-enable-plain-authentication.adoc [hunk 19]
```diff
@@ -0,0 +1,60 @@
+// Module included in the following assemblies:
+//
+// assembly-kafka-encryption-and-authentication.adoc
+
+[id='proc-kafka-enable-plain-authentication-{context}']
+
+= Enabling SASL PLAIN authentication
+
+This procedure will show you how to enable SASL PLAIN authentication in Kafka brokers.
```
Comment (dmuntima) [inline books/proc-kafka-enable-plain-authentication.adoc]: will show you how to > describes how to

File: books/proc-kafka-enable-plain-authentication.adoc [hunk 20]
```diff
@@ -0,0 +1,60 @@
...
+
+.Prerequisites
+
+* AMQ Streams is xref:proc-installing-amq-streams-{context}[installed] on all hosts which will be used as Kafka brokers.
+
+.Enabling SASL PLAIN authentication
```
Comment (dmuntima) [inline books/proc-kafka-enable-plain-authentication.adoc]: .Enabling SASL PLAIN authentication > .Procedure

File: books/proc-kafka-enable-plain-authentication.adoc [hunk 21]
```diff
@@ -0,0 +1,60 @@
...
++
+[source]
+----
+listeners=INSECURE://:9092,AUTHENTICATED://:9093,REPLICATION://:9094
+listener.security.protocol.map=INSECURE:PLAINTEXT,AUTHENTICATED:SASL_PLAINTEXT,REPLICATION:PLAINTEXT
+sasl.enabled.mechanisms=PLAIN
+----
+
+. (Re)start the Kafka brokers
+To pass the JAAS configuration to Kafka brokers, use the `KAFKA_OPTS` environment variable.
```
Comment (dmuntima) [inline books/proc-kafka-enable-plain-authentication.adoc]: To pass the  > to pass the

File: books/proc-kafka-enable-plain-authentication.adoc [hunk 22]
```diff
@@ -0,0 +1,60 @@
...
+----
+su - kafka
+export KAFKA_OPTS="-Djava.security.auth.login.config=/opt/kafka/config/jaas.conf"; /opt/kafka/bin/kafka-server-start.sh -daemon /opt/kafka/config/server.properties
+----
+
+.Additional resources
+
+* For more information about configuring AMQ Streams, see xref:proc-configuring-amq-streams-{context}[].
+* For more information about running a Kafka cluster, see xref:proc-running-multinode-kafka-cluster-{context}[].
+* For more information about configuring SASL PLAIN authentication in clients, see xref:producer-configuration-parameters-{context}[] and xref:consumer-configuration-parameters-{context}[].
```
Comment (dmuntima) [inline books/proc-kafka-enable-plain-authentication.adoc]: The output of the xrefs when I view the output of the file [1]. Please check.

[1] https://github.com/rh-messaging/amq-streams-docs/blob/d19c42537ae8c8e193f5619d99b9467c3dc18429/books/proc-kafka-enable-plain-authentication.adoc

File: books/proc-kafka-enable-scram-authentication.adoc [hunk 23]
```diff
@@ -0,0 +1,59 @@
...
+
+= Enabling SASL SCRAM authentication
+
+This procedure will show you how to enable SASL SCRAM authentication in Kafka brokers.
+
+.Prerequisites
+
+* AMQ Streams is xref:proc-installing-amq-streams-{context}[installed] on all hosts which will be used as Kafka brokers.
+
+.Enabling SASL SCRAM authentication
```
Comment (dmuntima) [inline books/proc-kafka-enable-scram-authentication.adoc]: .Enabling SASL SCRAM authentication > .Procedure

File: books/proc-kafka-enable-scram-authentication.adoc [hunk 24]
```diff
@@ -0,0 +1,59 @@
...
++
+[source]
+----
+listeners=INSECURE://:9092,AUTHENTICATED://:9093,REPLICATION://:9094
+listener.security.protocol.map=INSECURE:PLAINTEXT,AUTHENTICATED:SASL_PLAINTEXT,REPLICATION:PLAINTEXT
+sasl.enabled.mechanisms=SCRAM-SHA-512
+----
+
+. (Re)start the Kafka brokers
+To pass the JAAS configuration to Kafka brokers, use the `KAFKA_OPTS` environment variable.
```
Comment (dmuntima) [inline books/proc-kafka-enable-scram-authentication.adoc]: To pass the  > to pass the

File: books/proc-kafka-enable-scram-authentication.adoc [hunk 25]
```diff
@@ -0,0 +1,59 @@
...
+export KAFKA_OPTS="-Djava.security.auth.login.config=/opt/kafka/config/jaas.conf"; /opt/kafka/bin/kafka-server-start.sh -daemon /opt/kafka/config/server.properties
+----
+
+.Additional resources
+
+* For more information about configuring AMQ Streams, see xref:proc-configuring-amq-streams-{context}[].
+* For more information about running a Kafka cluster, see xref:proc-running-multinode-kafka-cluster-{context}[].
+* For more information about adding SASL SCRAM users, see xref:proc-kafka-adding-scram-users-{context}[].
+* For more information about deleting SASL SCRAM users, see xref:proc-kafka-deleting-scram-users-{context}[].
+* For more information about configuring SASL SCRAM authentication in clients, see xref:producer-configuration-parameters-{context}[] and xref:consumer-configuration-parameters-{context}[].
```
Comment (dmuntima) [inline books/proc-kafka-enable-scram-authentication.adoc]: The output of the xrefs is incorrect when I view the output of the file [1]. Please check.

[1] https://github.com/rh-messaging/amq-streams-docs/blob/d19c42537ae8c8e193f5619d99b9467c3dc18429/books/proc-kafka-enable-scram-authentication.adoc

File: books/proc-kafka-enable-tls-client-authentication.adoc [hunk 26]
```diff
@@ -0,0 +1,42 @@
...
+= Enabling TLS client authentication
+
+This procedure will show you how to enable TLS client authentication in Kafka brokers.
+
+.Prerequisites
+
+* AMQ Streams is xref:proc-installing-amq-streams-{context}[installed] on all hosts which will be used as Kafka brokers.
+* TLS encryption is xref:proc-kafka-enable-tls-encryption-{context}[enabled].
+
+.Enabling TLS client authentication
```
Comment (dmuntima) [inline books/proc-kafka-enable-tls-client-authentication.adoc]: .Enabling TLS client authentication > .Procedure

File: books/proc-kafka-enable-tls-client-authentication.adoc [hunk 27]
```diff
@@ -0,0 +1,42 @@
+// Module included in the following assemblies:
+//
+// assembly-kafka-encryption-and-authentication.adoc
+
+[id='proc-kafka-enable-tls-client-authentication-{context}']
+
+= Enabling TLS client authentication
+
+This procedure will show you how to enable TLS client authentication in Kafka brokers.
```
Comment (dmuntima) [inline books/proc-kafka-enable-tls-client-authentication.adoc]: will show you how to > describes how to

File: books/proc-kafka-enable-tls-client-authentication.adoc [hunk 28]
```diff
@@ -0,0 +1,42 @@
...
+ssl.client.auth=required
+----
+
+. (Re)start the Kafka brokers
+
+.Additional resources
+
+* For more information about configuring AMQ Streams, see xref:proc-configuring-amq-streams-{context}[].
+* For more information about running a Kafka cluster, see xref:proc-running-multinode-kafka-cluster-{context}[].
+* For more information about configuring TLS client authentication in clients, see xref:producer-configuration-parameters-{context}[] and xref:consumer-configuration-parameters-{context}[].
```
Comment (dmuntima) [inline books/proc-kafka-enable-tls-client-authentication.adoc]: The output of the xrefs is incorrect when I view the output of the file [1]. Please check.

[1] https://github.com/rh-messaging/amq-streams-docs/blob/d19c42537ae8c8e193f5619d99b9467c3dc18429/books/proc-kafka-enable-tls-client-authentication.adoc

File: books/proc-kafka-enable-tls-encryption.adoc [hunk 29]
```diff
@@ -0,0 +1,42 @@
...
+
+= Enabling TLS encryption
+
+This procedure will show you how to enable encryption in Kafka brokers.
+
+.Prerequisites
+
+* AMQ Streams is xref:proc-installing-amq-streams-{context}[installed] on all hosts which will be used as Kafka brokers.
+
+.Enabling TLS encryption
```
Comment (dmuntima) [inline books/proc-kafka-enable-tls-encryption.adoc]: .Enabling TLS encryption > .Procedure

File: books/proc-kafka-enable-tls-encryption.adoc [hunk 30]
```diff
@@ -0,0 +1,42 @@
+// Module included in the following assemblies:
+//
+// assembly-kafka-encryption-and-authentication.adoc
+
+[id='proc-kafka-enable-tls-encryption-{context}']
+
+= Enabling TLS encryption
+
+This procedure will show you how to enable encryption in Kafka brokers.
```
Comment (dmuntima) [inline books/proc-kafka-enable-tls-encryption.adoc]: will show you how to > describes how to

File: books/proc-kafka-enable-tls-encryption.adoc [hunk 31]
```diff
@@ -0,0 +1,42 @@
...
+listener.security.protocol.map=UNENCRYPTED:PLAINTEXT,ENCRYPTED:SSL,REPLICATION:PLAINTEXT
+ssl.keystore.location=/path/to/keystore/server-1.jks
+ssl.keystore.password=123456
+----
+
+. (Re)start the Kafka brokers
+
+.Additional resources
+
+* For more information about configuring AMQ Streams, see xref:proc-configuring-amq-streams-{context}[].
```
Comment (dmuntima) [inline books/proc-kafka-enable-tls-encryption.adoc]: The output of the xrefs is incorrect when I view the output of the file [1]. Please check.

[1] https://github.com/rh-messaging/amq-streams-docs/blob/d19c42537ae8c8e193f5619d99b9467c3dc18429/books/proc-kafka-enable-tls-encryption.adoc

File: books/proc-kafka-deleting-scram-users.adoc [hunk 32]
```diff
@@ -0,0 +1,30 @@
...
+bin/kafka-configs.sh --zookeeper _<ZookeeperAddress>_ --alter --delete-config 'SCRAM-SHA-512' --entity-type users --entity-name _<Username>_
++
+For example:
++
+[source,subs=+quotes]
+bin/kafka-configs.sh --zookeeper zoo1.my-domain.com:2181 --alter --delete-config 'SCRAM-SHA-512' --entity-type users --entity-name user1
+
+.Additional resources
+
+* For more information about configuring SASL SCRAM authentication in clients, see xref:producer-configuration-parameters-{context}[] and xref:consumer-configuration-parameters-{context}[].
```
Comment (dmuntima) [inline books/proc-kafka-deleting-scram-users.adoc]: The output of the xrefs when I view the output of the file [1]. Please check.

[1] https://github.com/rh-messaging/amq-streams-docs/blob/d19c42537ae8c8e193f5619d99b9467c3dc18429/books/proc-kafka-deleting-scram-users.adoc

File: books/proc-kafka-adding-scram-users.adoc [hunk 33]
```diff
@@ -0,0 +1,30 @@
...
+bin/kafka-configs.sh --zookeeper _<ZookeeperAddress>_ --alter --add-config 'SCRAM-SHA-512=[password=_<Password>_]' --entity-type users --entity-name _<Username>_
++
+For example:
++
+[source,subs=+quotes]
+bin/kafka-configs.sh --zookeeper zoo1.my-domain.com:2181 --alter --add-config 'SCRAM-SHA-512=[password=123456]' --entity-type users --entity-name user1
+
+.Additional resources
+
+* For more information about configuring SASL SCRAM authentication in clients, see xref:producer-configuration-parameters-{context}[] and xref:consumer-configuration-parameters-{context}[].
```
Comment (dmuntima) [inline books/proc-kafka-adding-scram-users.adoc]: The output of the xrefs when I view the output of the file [1]. Please check.

[1] https://github.com/rh-messaging/amq-streams-docs/blob/d19c42537ae8c8e193f5619d99b9467c3dc18429/books/proc-kafka-adding-scram-users.adoc

File: books/con-kafka-authentication.adoc [hunk 34]
```diff
@@ -0,0 +1,210 @@
...
+
+= Authentication
+
+Kafka supports two methods of authentication.
+On all conenctions, authentication using one of the supported SASL (Simple Authentication and Security Layer) mechanisms can be used.
+On encrypted connections, TLS client authentication based on X.509 certificates can be used.
+
+== TLS cLient authentication
+
+TLS CLient authentication can be used only on connections which are already using TLS encryption.
```
[the remark this answers is not in the document]
Comment (tombentley) [reply books/con-kafka-authentication.adoc]: Also, it should be "client" (lowercase), right?

[the remark this answers is not in the document]
Comment (dmuntima) [reply books/con-kafka-authentication.adoc]: oh yeah, "client" (lowercase).

File: books/proc-kafka-enable-plain-authentication.adoc (hunk 21 above)
Comment (tombentley) [reply books/proc-kafka-enable-plain-authentication.adoc]: That doesn't quite read right to me; the way the ", use the `KAFKA_OPTS` environment variable" is added to the end of the sentence. How about:

> . (Re)start the Kafka brokers using the `KAFKA_OPTS` environment variable to pass the JAAS configuration to Kafka brokers.

Comment (dmuntima) [reply books/proc-kafka-enable-plain-authentication.adoc]: This sounds good.

File: books/proc-kafka-enable-plain-authentication.adoc (hunk 22 above)
Comment (tombentley) [reply books/proc-kafka-enable-plain-authentication.adoc]: This renders fine for me when I use asciidoctor.

File: books/proc-kafka-enable-scram-authentication.adoc (hunk 25 above)
Comment (tombentley) [reply books/proc-kafka-enable-scram-authentication.adoc]: Again, this seems fine with asciidoctor.

File: books/proc-kafka-enable-tls-client-authentication.adoc (hunk 28 above)
Comment (tombentley) [reply books/proc-kafka-enable-tls-client-authentication.adoc]: Again, this seems fine with asciidoctor.

**Comment by tombentley** [comment]
Rebased for conflict and addressed review comments. @dmuntima are you happy with the comments I've made about the review?

**Comment by dmuntima** [comment]
Thanks, @tombentley. It looks good.

**Pull request merged**